discovery-provider/src/api_helpers.py

```diff
@@ -33,15 +33,15 @@ def error_response(error, error_code=500):
 def success_response_backwards_compat(response_entity=None, status=200):
     response_dictionary = response_dict_with_metadata(response_entity)
     response_dictionary = {**response_dictionary, **response_entity}
-    signature, timestamp = generate_signature_and_timestamp({'data': response_dictionary['data']})
+    signature, timestamp = generate_signature_and_timestamp(response_dictionary)
     response_dictionary['signature'] = signature
     response_dictionary['timestamp'] = timestamp
     return jsonify(response_dictionary), status
 
 # Create a response dict with metadata, data, signature, and timestamp
 def success_response(response_entity=None, status=200):
```

Comment:
imo this could be cleaned up a bit - can response_dict_with_metadata add the timestamp and do the signature in one self-contained thing? seems a bit messy to use a function to edit the object, generate a signature by editing the object again, then edit the object a third time because we don't get the edited object back from the second time